compiler-core/src/docs/printer.rs

```diff
@@ -0,0 +1,584 @@
+                    ))
+                }
+            }
+            Type::Fn { args, return_ } => docvec![
+                self.keyword("fn"),
+                Self::type_arguments(args.iter().map(|argument| self.type_(argument))),
+                " -> ",
+                self.type_(return_)
+            ],
+            Type::Tuple { elements } => docvec![
+                "#",
+                Self::type_arguments(elements.iter().map(|element| self.type_(element))),
+            ],
+            Type::Var { type_ } => match type_.as_ref().borrow().deref() {
+                TypeVar::Link { type_ } => self.type_(type_),
+
+                TypeVar::Unbound { id } | TypeVar::Generic { id } => {
+                    let name = self.type_variable(*id);
+                    self.variable(name)
+                }
+            },
+        }
+    }
+
+    fn type_variable(&mut self, id: u64) -> EcoString {
+        if let Some(name) = self.names.get_type_variable(id) {
+            return name.clone();
+        }
+
+        if let Some(name) = self.printed_type_variables.get(&id) {
+            return name.clone();
+        }
+
+        loop {
+            let name = self.next_letter();
+            if !self.printed_type_variable_names.contains(&name) {
+                _ = self.printed_type_variable_names.insert(name.clone());
+                _ = self.printed_type_variables.insert(id, name.clone());
+                return name;
+            }
+        }
+    }
+
+    // Copied from the `next_letter` method of the `type_::printer`.
+    fn next_letter(&mut self) -> EcoString {
+        let alphabet_length = 26;
+        let char_offset = b'a';
+        let mut chars = vec![];
+        let mut n;
+        let mut rest = self.next_type_variable_id;
+
+        loop {
+            n = rest % alphabet_length;
+            rest = rest / alphabet_length;
+            chars.push((n as u8 + char_offset) as char);
+
+            if rest == 0 {
+                break;
+            }
+            rest -= 1
+        }
+
+        self.next_type_variable_id += 1;
+        chars.into_iter().rev().collect()
+    }
+
+    fn named_type_name(&self, package: &str, module: &str, name: &EcoString) -> Document<'static> {
+        if package == PRELUDE_PACKAGE_NAME && module == PRELUDE_MODULE_NAME {
+            self.title(name)
+        } else if package == self.package && module == self.module {
+            self.link(eco_format!("#{name}"), self.title(name), None)
+        } else {
+            let module_name = module.split('/').next_back().unwrap_or(module);
+            let qualified_name = docvec![
+                self.variable(EcoString::from(module_name)),
+                ".",
+                self.title(name)
+            ];
+            let title = eco_format!("{module}.{{type {name}}}");
+
+            self.link(
+                eco_format!("https://hexdocs.pm/{package}/{module}.html#{name}"),
```

Comment:
I think this link will be wrong in these situations:

- If the module is from the root package and viewing the docs locally
- For paths or git deps, so not on Hex
- For deps that are not the most latest version of that package on Hex

For path and git deps I think we'll have to offer no link, for the root we'll need to use relative links, and the Hex one we'll need to use the current version.